source/tests/unit/xnet_converters_tests.cpp

```diff
@@ -599,6 +599,103 @@ TEST(XnetConvertersTests, IPv4AddrOutputConverter_ConvertToGrpc_ConvertsToAddres
+  constexpr auto NEXT_PORT = 101;
+  constexpr auto NEXT_ADDR = 1234568;
+  nxsockaddr_in next_addr{nxAF_INET, NEXT_PORT, {NEXT_ADDR}};
+  nxsockaddr_in addr{nxAF_INET, PORT, {ADDR}};
+  nxaddrinfo addr_info_next{
+      nxAI_PASSIVE | nxAI_ADDRCONFIG,
+      nxAF_INET,
+      nxSOCK_DGRAM,
+      nxIPPROTO_UDP,
+      (nxsocklen_t)sizeof(next_addr),
+      reinterpret_cast<nxsockaddr*>(&next_addr),
+      "NextAddrHostName",
+      nullptr};
+  nxaddrinfo addr_info{
+      nxAI_ALL,
+      nxAF_INET,
+      nxSOCK_STREAM,
+      nxIPPROTO_TCP,
+      (nxsocklen_t)sizeof(addr),
+      reinterpret_cast<nxsockaddr*>(&addr),
+      "AddrHostName",
+      &addr_info_next};
+
+  NiXnetSocketMockLibrary library;
+  auto converter = allocate_output_storage<nxaddrinfo, google::protobuf::RepeatedPtrField<AddrInfo>>(&library);
+  converter.addr_info_ptr = &addr_info;
+
+  EXPECT_CALL(library, FreeAddrInfo(converter.addr_info_ptr))
+      .Times(1);
+  google::protobuf::RepeatedPtrField<nixnetsocket_grpc::AddrInfo> grpc_output;
+  convert_to_grpc(converter, &grpc_output);
+
+  EXPECT_EQ(2, grpc_output.size());
+  auto first_addr_info = grpc_output[0];
+  EXPECT_EQ(nxAI_ALL, first_addr_info.flags_raw());
+  EXPECT_EQ(1, first_addr_info.flags().size());
+  EXPECT_EQ(GetAddrInfoFlags::GET_ADDR_INFO_FLAGS_ALL, first_addr_info.flags(0));
+  EXPECT_EQ(AddressFamily::ADDRESS_FAMILY_INET, first_addr_info.family());
+  EXPECT_EQ(SocketProtocolType::SOCKET_PROTOCOL_TYPE_STREAM, first_addr_info.sock_type());
+  EXPECT_EQ(IPProtocol::IP_PROTOCOL_TCP, first_addr_info.protocol());
+  EXPECT_STREQ("AddrHostName", first_addr_info.canon_name().c_str());
+  EXPECT_EQ(ADDR, first_addr_info.addr().ipv4().addr().addr());
+  EXPECT_EQ(PORT, first_addr_info.addr().ipv4().port());
+  auto next_addr_info = grpc_output[1];
+  EXPECT_EQ(nxAI_PASSIVE | nxAI_ADDRCONFIG, next_addr_info.flags_raw());
+  EXPECT_EQ(2, next_addr_info.flags().size());
+  EXPECT_EQ(GetAddrInfoFlags::GET_ADDR_INFO_FLAGS_ADDRCONFIG, next_addr_info.flags(0));
+  EXPECT_EQ(GetAddrInfoFlags::GET_ADDR_INFO_FLAGS_PASSIVE, next_addr_info.flags(1));
+  EXPECT_EQ(AddressFamily::ADDRESS_FAMILY_INET, next_addr_info.family());
+  EXPECT_EQ(SocketProtocolType::SOCKET_PROTOCOL_TYPE_DGRAM, next_addr_info.sock_type());
+  EXPECT_EQ(IPProtocol::IP_PROTOCOL_UDP, next_addr_info.protocol());
+  EXPECT_STREQ("NextAddrHostName", next_addr_info.canon_name().c_str());
+  EXPECT_EQ(NEXT_ADDR, next_addr_info.addr().ipv4().addr().addr());
+  EXPECT_EQ(NEXT_PORT, next_addr_info.addr().ipv4().port());
+}
+
+TEST(XnetConvertersTests, AddrInfoFlagsAsInt_ConvertToRepeatedFlagsEnum_AllFlagsPresent)
+{
+  int32_t flags = nxAI_BYPASS_CACHE | nxAI_V4MAPPED | nxAI_PASSIVE;
+
+  AddrInfo grpc_addr_info{};
+  convert_to_addr_info_flags(flags, *(grpc_addr_info.mutable_flags()));
+
+  EXPECT_EQ(3, grpc_addr_info.flags().size());
+  EXPECT_EQ(GetAddrInfoFlags::GET_ADDR_INFO_FLAGS_BYPASS_CACHE, grpc_addr_info.flags(0));
```

Comment:
Question: I may have missed it but how are these flags ordered?

Reply:
They're in descending order right now so you might get (2048, 4, 1). I can change the conversion code to have it in ascending if it's useful. I'm not sure if there's an order of importance to the flags.

Reply:
I added AB#1900318 to the API review story to come back to this in general for any repeated flag enum fields to evaluate if the order's important and update accordingly.